lib/api/objectPutPart.js

```diff
@@ -118,36 +121,37 @@ export default function objectPutPart(authInfo, request, streamingV4Params, log,
                             'the destination bucket', {
                                 error: err,
                             });
-                        return next(err, bucket);
+                        return next(err, destinationBucket);
                     }
-                    return next(null, res, bucket);
+                    return next(null, destinationBucket, res);
                 });
             }
             // The bucket does not have server-side encryption, so pass `null`
-            return next(null, null, bucket);
+            return next(null, destinationBucket, null);
         },
         // Get the MPU shadow bucket.
-        (cipherBundle, bucket, next) => metadata.getBucket(mpuBucketName, log,
+        (destinationBucket, cipherBundle, next) =>
```

Comment:
This kind of call chains with quite a few order-dependent arguments would benefit from passing them as object and using deconstruction to get them in callees (this is more or a general cleanup that can be the topic of another PR, but it would have helped here I think especially when moving arguments around).